tests/test_reduction_ops.py

```diff
@@ -812,19 +812,42 @@ def test_accuracy_conv2d(shape, kernel, stride, padding, groups, dtype, dilation
     res_out = flag_gems.conv2d(
         inp,
         weight,
-        bias=None,
+        bias=bias,
         groups=groups,
         stride=stride,
         padding=padding,
         dilation=dilation,
     )
+
     gems_assert_close(res_out, ref_out, dtype)
+
     out_grad = torch.randn_like(ref_out).to(flag_gems.device)
+
     ref_grad = to_reference(out_grad, True)
-    (ref_in_grad,) = torch.autograd.grad(ref_out, ref_inp, ref_grad)
-    (res_in_grad,) = torch.autograd.grad(res_out, inp, out_grad)
+    if bias is not None:
+        (ref_in_grad, ref_weight_grad, ref_bias_grad) = torch.autograd.grad(
+            ref_out, (ref_inp, ref_weight, bias_ref), ref_grad
+        )
+        (res_in_grad, res_weight_grad, res_bias_grad) = torch.autograd.grad(
+            res_out, (inp, weight, bias), out_grad
+        )
+    else:
+        (ref_in_grad, ref_weight_grad) = torch.autograd.grad(
+            ref_out, (ref_inp, ref_weight), ref_grad
+        )
+        (res_in_grad, res_weight_grad) = torch.autograd.grad(
+            res_out, (inp, weight), out_grad
+        )
 
-    gems_assert_close(res_in_grad, ref_in_grad.to(dtype), dtype)
+    gems_assert_close(
+        res_in_grad.to(dtype), ref_in_grad.to(dtype), dtype, reduce_dim=weight.shape[2]
```

Comment:
casting is useless here. gems_assert_close will do type casting for reference tensors.